src/integrations/admin/indexation-integration.php

```diff
@@ -149,47 +170,19 @@ public function enqueue_scripts() {
 			return;
 		}
 
-		\add_action( 'admin_footer', [ $this, 'render_indexation_modal' ], 20 );
+		$this->indexation_action_is_link = ! $this->yoast_tools_page_conditional->is_met();
```

Comment:
You probably can set this in the constructor, because the conditionals are loaded before the classes are instantiated.

Also this kind of assignments are hard to read (for me personally). It should be a link, but only when it is not the tools page. Why not reversing this: `indexation_action_is_button`. And also in this particular file you are using the variable especially to check if it is not the tools page -> why not naming this: 'is_tools_page'. 

An alternative is passing the result of `$this->yoast_tools_page_conditional->is_met()`. This  means that you don't need a variable anymore that is named: `indexation_action_is_link`.

Reply:
See other convo 😉